import_flow_events.py

```diff
@@ -100,12 +101,15 @@
 """
 
 Q_CLEAR_DAY_METADATA = """
-    DELETE FROM flow_metadata
-    WHERE begin_time::DATE = '{day}'::DATE;
+    DELETE m
+    FROM flow_metadata AS m
+    LEFT JOIN flow_events AS e
+      ON m.flow_id = e.flow_id
+    WHERE e.flow_id IS NULL;
```

Comment:
> ...this will work well for the case where you're re-importing the later half of a flow, but I'm not sure about when you're importing the earlier half (i.e. you're deleting and re-importing the flow.begin event, but not re-importing the events that complete it).

I've been thinking about this overnight and the only sane approach I can think of involves special-casing the `INSERT` to `flow_metadata`, so that it checks for the existence of that `flow_id` before inserting. I'm going to play around with this approach a bit now, then I'll come back and update the PR (or not).